databricks/resource_databricks_workspace_conf.go

```diff
@@ -0,0 +1,198 @@
+package databricks
+
+// Preview feature: https://docs.databricks.com/security/network/ip-access-list.html
+// REST API: https://docs.databricks.com/dev-tools/api/latest/ip-access-list.html#operation/create-list
+
+import (
+	"strconv"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceWorkspaceConf() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWorkspaceConfCreateOrUpdate,
+		Read:   resourceWorkspaceConfRead,
+		Update: resourceWorkspaceConfCreateOrUpdate,
+		Delete: resourceWorkspaceConfDelete,
+
+		Schema: map[string]*schema.Schema{
+			"enable_ip_access_lists": {
```

Comment:
please rewrite to `schema.TypeMap` unless you have explicit OK from all PMs to make these properties public :) with this approach we can properly combine public & private configurations (which may change it's format)

Reply:
Based on the API, this will always be a map.  I might need a little hint, here.  Delete logic for each key is going to be dependent on the key.  enableIpAccessLists is in the public docs, so I was thinking this was ok.

Reply:
I think what he might be saying is that the field is something like "configuration" where the type is TypeMap of all string values. The user can the pick and choose what workspace conf items get added. This obscures all the "keys" available but I am not sure if I totally agree with that approach as we cant take advantage of schema constructs such as validation, etc.

Reply:
Right, what do you think @nfx ?  We only add features here that show up in the public docs.  Each option has some special handling around how you delete and such.  I think we'll need to keep updating this as the platform keeps using.  i put special code in this resource based on the specific option of "enable_ip_access_lists" in this case.

Reply:
well... in order not to maintain "public" & "private" forks, we need to make these flexible. so, instead of `enable_ip_access_list` we should have `features = ["ip_access_lists"]` for "workspace feature flags"